autoload/go/package.vim

```diff
@@ -82,13 +86,21 @@ function! s:vendordirs() abort
     if l:err != 0
       return []
     endif
-    let l:root = split(l:root, '\n')[0] . go#util#PathSep() . 'src'
+    let l:lines = split(l:root, '\n')
+    if empty(l:lines)
+      return []
+    endif
+    let l:root = l:lines[0] . go#util#PathSep() . 'src'
 
     let [l:dir, l:err] = go#util#ExecInDir(['go', 'list', '-f', '{{.Dir}}'])
     if l:err != 0
       return []
     endif
-    let l:dir = split(l:dir, '\n')[0]
+    let l:lines = split(l:dir, '\n')
+    if empty(l:lines)
+      return {}
+    endif
+    let l:dir = l:lines[0]
```

Comment:
Why is this second new block necessary? Assuming that `go list` in a null modules exits 0, is there some condition where `.Root` will have a value, but `.Dir` will not?

Reply:
No. go list return `.` not null.

![image](https://user-images.githubusercontent.com/10111/58857136-73320e00-86df-11e9-9938-303bbf689da2.png)

Reply:
go list possibly return empty and 0.

![image](https://user-images.githubusercontent.com/10111/58857415-2f8bd400-86e0-11e9-8032-aa3abcc76bfe.png)

Reply:
Oops. Sorry. This is my fault.